extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -267,35 +273,27 @@ public void withHint(String hintName, Object value) {
 
     // Results
 
-    @SuppressWarnings("unchecked")
     public long count() {
         if (count == null) {
```

Comment:
Unrelated to this PR, but caching the `count` could lead to really nasty surprises for users... not sure it's a great idea.

Reply:
I often wondered about this. Funny that nobody reported issues with this. This is mostly useful because any calls to paging info (number of pages, etc) would cause extra calls to `count()` but I wonder if ORM caches that itself, perhaps I don't need to cache.

Reply:
Hibernate ORM doesn't cache counts by default AFAICS. It can probably cache them, but that would be through the same mechanism as other queries (`Query#setCacheable`) -- and in the second-level cache, so would affect other sessions.

I suppose a first-level cache for query execution is an interesting ideas for some scenarios, such as query counts. Maybe we should open a feature request.

Reply:
Honestly, I don't know how important that is. Or how usable/realistic.